### ISSUE TITLE
flutter version fails to switch to version

### ISSUE THREAD

**Issue by peterlauri** [flutter/flutter#55383]
I wanted to try out 1.18.0-dev.0.0, but there looks to be a bug in the **flutter version** command, as it complains that it cannot find 1.18.0-de.0.0 (**note the missing [v] in dev**)...

```
➜  image_network_problem git:(master) ✗ flutter version
1.18.0-6.0.pre
1.18.0-dev.5.0
1.18.0-dev.4.0
1.18.0-dev.3.0
1.17.0-dev.3.1
1.18.0-dev.2.0
1.18.0-dev.1.0
1.18.0-dev.0.0
1.17.0-dev.5.0
1.17.0-dev.4.0
1.17.0-dev.3.0
... truncated ...
v0.0.20-alpha
➜  image_network_problem git:(master) ✗ flutter version 1.18.0-dev.0.0
╔══════════════════════════════════════════════════════════════════════════════╗
║ Warning: "flutter version" will leave the SDK in a detached HEAD state.      ║
║ If you are using the command to return to a previously installed SDK version ║
║ consider using the "flutter downgrade" command instead.                      ║
╚══════════════════════════════════════════════════════════════════════════════╝


Are you sure you want to proceed? [y|n]: y
There is no version: 1.18.0-de.0.0
Unable to checkout version branch for version 1.18.0-de.0.0: ProcessException: Process exited
abnormally:
error: pathspec '1.18.0-de.0.0' did not match any file(s) known to git
  Command: git checkout 1.18.0-de.0.0
```

Flutter doctor -v:

```
➜  image_network_problem git:(master) ✗ flutter doctor -v
[✓] Flutter (Channel beta, v1.17.0, on Mac OS X 10.15.3 19D76, locale en-GB)
    • Flutter version 1.17.0 at /Users/plauri/.applications/flutter
    • Framework revision d3ed9ec945 (2 weeks ago), 2020-04-06 14:07:34 -0700
    • Engine revision c9506cb8e9
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 eea9717938)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/plauri/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

 
[✓] Connected device (4 available)            
    • PRA LX1             • 8NH7N17B04000425 • android-arm64  • Android 8.0.0 (API 26)
    • AOSP on IA Emulator • emulator-5554    • android-x86    • Android 9 (API 28) (emulator)
    • Chrome              • chrome           • web-javascript • Google Chrome 81.0.4044.113
    • Web Server          • web-server       • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

**Comment by peterlauri** [comment]
@VladyslavBondarenko PR is ready... even I struggled with formatting :/

**Comment by christopherfujino** [comment]
Closing this as a duplicate of https://github.com/flutter/flutter/issues/55421 for tracking purposes. In the end it doesn't make a big difference, as once the tree is green I will land your PR and resolve both.
